packages/netlify/src/index.ts

```diff
@@ -155,6 +155,7 @@ export default function netlifyIntegration(
 			entryPoints: [fileURLToPath(new URL('./entry.mjs', middlewareOutputDir()))],
 			target: 'es2022',
 			platform: 'neutral',
+			mainFields: [ 'main' ],
```

Comment:
I believe this would ignore packages with a valid `module` field, instead pulling in the code from `main`? I think we want this instead:

```suggestion
			mainFields: [ 'module', 'main' ],
```

Reply:
I found out about this config when bluwy mentioned it. From what I can tell from esbuild docs ([Main Fields](https://esbuild.github.io/api/#main-fields)) , `"module"` is intended for code targeting browsers. Would we want esbuild to consider it for middleware?

Reply:
I think we should add the `module` field here. The esbuild docs mention that people tend to use `module` for browser-targetting envs, but it's not what `module` field was meant to be. They should use the `browser` field instead.

Reply:
There's more docs on this in the [Platform section](https://esbuild.github.io/api/#platform) of ESBuild docs:

> When the platform is set to neutral:
> ....
> The [main fields](https://esbuild.github.io/api/#main-fields) setting is empty by default. If you want to use npm-style packages, you will likely have to configure this to be something else such as `main` for the standard `main` field used by node.

That makes me think we should use `["main", "module"]`, in that order, because ESBuild will use the array ordering as the precedence order. We're bundling Edge Functions here, which run in an environment that's more similar to Node.js than it is to Browsers. So if `module` sometimes is used for browser builds, we should prefer the Node.js `main` build over that.

Having said that, I don't have too strong of an opinion on the ordering here. But I do agree that we should include both, yes!

Reply:
main then module sounds good to me, updating the suggestion to add it.

Reply:
Usually when both main and module exist, module should take precedence first, so I'm not really sure flipping that is a good idea.

Reply:
Is it a common config?

Reply:
Yeah, it's common for module and main to be set together. And usually module comes first, then main. `main` is like the "universal fallback" that most toolings know, so having module take precedence and be the more specific alternative makes more sense.

Thanks for making the change!

Reply:
Thanks. If it's common, I don't think we don't need reconsider it.